tsconfig.json

```diff
@@ -10,7 +10,10 @@
         "sourceMap": true,
         "resolveJsonModule": true,
         "esModuleInterop": true,
-        "noEmit": true,
+        "noEmit": false,
+        "emitDeclarationOnly": true,
+        "declaration": true,
+        "outFile": "dist/fp-toolkit",
```

Comment:
Emit declaration files so we still get TS auto complete and bundle them all into one `fp-toolkit.d.ts` file in `dist`.

Reply:
After reviewing below, I wonder if this was somehow related to typedoc's compilation competing? 🤔

Reply:
>  I got a bunch of weird type naming conflicts/errors.

Was this at build time of the library or when you tried to consume it? I haven't done that just yet.